pkg/sinit/init.go

```diff
@@ -145,153 +145,164 @@ type LogrusConfig struct {
 	ExitOnFatal bool
 }
 
-var initOnce sync.Once
-var handlerMulti *strc.MultiHandler
-var handlerSplunk *splunk.SplunkHandler
-var handlerCloudWatch *cloudwatchwriter2.Handler
+type resources struct {
+	handlerMulti      *strc.MultiHandler
+	handlerSplunk     *splunk.SplunkHandler
+	handlerCloudWatch *cloudwatchwriter2.Handler
+	sentryEnabled     bool
+}
+
+var (
+	res                     *resources
+	resMu                   sync.Mutex
+
+	ErrAlreadyInitialized   = errors.New("logging already initialized, call Close clean")
+	ErrInvalidURL           = errors.New("invalid URL")
+	ErrSentryInitialization = errors.New("sentry initialization error")
+)
 
-// InitializeLogging initializes the logging system with the provided configuration. Use Flush to ensure all logs are written before exiting.
-// Subsequent calls to InitializeLogging will have no effect and will not return any error.
+// InitializeLogging initializes the logging system with the provided
+// configuration. Use Close to ensure all logs are written before exiting.
+// Subsequent calls to InitializeLogging will lead to ErrAlreadyInitialized.
 func InitializeLogging(ctx context.Context, config LoggingConfig) error {
-	var outerError error
+	resMu.Lock()
+	defer resMu.Unlock()
 
-	initOnce.Do(func() {
-		var handlers []slog.Handler
+	if res != nil {
+		return ErrAlreadyInitialized
+	}
+	res = &resources{}
```

Comment:
Nothing was changed in the "once" block after this line, the indentation change hunk is a bit hard to read.